modules/menu/Menu.tsx

```diff
@@ -0,0 +1,81 @@
+import React, { useMemo } from "react";
+import {
+  StandardProps,
+  WithStylesProps,
+  ThemeStyleSheetFactory
+} from "@diana-ui/types";
+import { withStyles } from "@diana-ui/base";
+import { Icon, IconNames } from "@diana-ui/icon";
+import { IPopoverProps, Popover, usePopoverRef } from "@diana-ui/popover";
+
+export interface IProps extends IPopoverProps {
+  menuIcon: IconNames;
+}
+
+const styleSheet: ThemeStyleSheetFactory = theme => ({
+  menuIcon: {},
+  list: {
+    listStyle: "none",
+    marginBottom: 0,
+    marginTop: 0,
+    maxHeight: 300,
+    overflowY: "auto",
+    paddingLeft: 0,
+    width: "100%"
+  },
+  item: {
+    cursor: "pointer"
+  }
+});
+
+export const styleSheetPopover: ThemeStyleSheetFactory = () => ({
+  container: {
+    maxWidth: "100%",
+    minWidth: 220
+  }
+});
+
+const StyledPopover = Popover.extendStyles(styleSheetPopover);
+
+const Menu: React.FC<IProps & WithStylesProps> = ({
+  className,
+  children,
+  wrappedRef,
+  cx,
+  styles,
+  menuIcon,
+  parentStylesheet,
+  ...props
+}) => {
+  const ref = usePopoverRef(wrappedRef);
+  const hide = () => ref.current?.hide();
+
+  const renderCustomHeader = useMemo(
+    () => () => <Icon name={menuIcon} className={cx(styles.menuIcon)} />,
+    [cx, menuIcon, styles]
+  );
+
+  return (
+    <StyledPopover
+      {...props}
+      className={className}
+      wrappedRef={ref}
+      renderHeader={renderCustomHeader}
+    >
+      <ul className={cx(styles.list, "list")}>
+        {React.Children.map(children, (item, index) => (
+          <li
```

Comment:
Why is every `MenuItem` a `div` inside a `li`? Couldn't we just have the `li`? IMO there is nothing being handled here that couldn't be handled in `MenuItem`

Reply:
Not quite folllowing. Are you saying we should have the li inside the menuItem?

Reply:
Yes - I think we only need to pass down onHide as a prop when mapping the children, the rest can be handled by inside `MenuItem`.
Am I missing something that makes us need the <li> <div /> </li> structure? What do you think?

Reply:
I think that maybe we shouldn't put the `<li>` inside.. because it works on Menu with a `<ul>` but if we get to use a MenuItem somewhere else it will not make sense